src/algebra/z64/batch.rs

```diff
@@ -0,0 +1,139 @@
+use super::*;
+
+use crate::util::Writer;
+
+use itertools::izip;
+use serde::{Deserialize, Serialize};
+
+use std::convert::TryInto;
+use std::io;
+
+pub(super) const BATCH_SIZE: usize = 1;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Batch(pub(super) [u64; BATCH_SIZE]);
+
+impl Add for Batch {
+    type Output = Self;
+
+    #[allow(clippy::suspicious_arithmetic_impl)]
+    #[inline(always)]
+    fn add(self, other: Self) -> Self::Output {
+        let mut res: [u64; BATCH_SIZE] = [0; BATCH_SIZE];
```

Comment:
This *probably* optimizes just fine since Rust can see that `BATCH_SIZE` is always 1, but you could probably erase this loop entirely and just do something roughly like for the function body:

```rust
res[0] = self.0[0] + other.0[0];
Self(res)
```

Same with the other math op impls (`Sub`, `Mul`, etc).

Reply:
Totally. I left it general because I wasn't sure if we'd ever have a use for larger batches of Z64 elements, but if the answer is definitely no I can go back and make everything 1-dimensional

Reply:
> I left it general because I wasn't sure if we'd ever have a use for larger batches of Z64 elements,

Makes sense. I'm definitely not qualified to know whether we will 😅. It's probably fine to leave as-is, since I know for a fact that Rust is smart enough to optimize the other `izip` uses into single 64-bit math ops.

Reply:
@rot256 what do you think about this - should we leave batch size for Z_64 flexible or hard-code it to be 1?

Reply:
There is nothing wrong with fixing the batch size to 1.
The only reason to have a larger batch size would be SIMD operations.